ThunderCloud/TSCAppDelegate.swift

```diff
@@ -11,30 +11,51 @@ import UserNotifications
 import ThunderBasics
 import ThunderRequest
 import ThunderTable
+import Baymax
 import CoreSpotlight
+import BackgroundTasks
 
 @UIApplicationMain
 /// A root app delegate which sets up your window and push notifications e.t.c.
 open class TSCAppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate {
 
 	/// The main window of the app
 	open var window: UIWindow?
+    
+    /// A window for presenting login UI
+    private var loginWindow: UIWindow?
 	
 	/// Whether to show push notifications when the app is in the foreground
 	public var foregroundNotificationOptions: UNNotificationPresentationOptions? = [.alert, .badge, .sound]
+    
+    static let appStateCategory = "AppState"
 	
     open func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
-		
+        
+        baymax_log("application:DidFinishLaunchingWithOptions with keys: \(launchOptions?.keys.map({ $0.rawValue }).description ?? "[]")", subsystem: Logger.stormSubsystem, category: TSCAppDelegate.appStateCategory, type: .info)
+                
 		UNUserNotificationCenter.current().delegate = self
-		
-		window = UIWindow(frame: UIScreen.main.bounds)
-		
-		let appVCClass: AppViewController.Type = StormObjectFactory.shared.class(for: String(describing: AppViewController.self)) as? AppViewController.Type ?? AppViewController.self
-		window?.rootViewController = appVCClass.init()
-		window?.makeKeyAndVisible()
-		
+				
+        setupRootWindow()
 		setupSharedUserAgent()
         
+        if let remoteNotification = launchOptions?[.remoteNotification] as? [String : Any], let aps = remoteNotification["aps"] as? [AnyHashable : Any] {
+            
+            baymax_log("App was launched by remote notification:\n\(String(remoteNotification) ?? "Unable to Parse")", subsystem: Logger.stormSubsystem, category: "PushNotifications", type: .info)
+            let launchedByContentPush = aps.keys.count == 1 && aps["content-available"] as? Int == 1
```

Comment:
Why is there necessarily only 1 key in `aps` if `content-available`. Feel like the APNS payload can still add stuff to `aps` if they want? Or is this an agreed Storm format

Reply:
Will change!